src/pages/map/map.spec.ts

```diff
@@ -34,11 +37,19 @@ describe('MapPage', function () {
       getCurrentPosition: stub().returns(geolocationDeferred.promise)
     };
 
+    popoverMock = {
+      present: stub().returns(new Deferred().promise)
```

Comment:
Using `Deferred` is unnecessary unless you need the deferred object to manually resolve/reject it later. You could replace `new Deferred().promise` with `new Promise()`. And Sinon actually simplifies this to `stub().resolves()` (or `stub().resolves(value)` if you want the promise to be resolved with `value`).